src/LibraryManager.Vsix/Shared/VsHelpers.cs

```diff
@@ -236,6 +237,60 @@ public static Project GetDTEProject(IVsHierarchy hierarchy)
 
             return null;
         }
+
+        public static string GetProjectName(string filePath)
+        {
+            if (String.IsNullOrEmpty(filePath))
+            {
+                return null;
+            }
+
+            string projectName = null;
+            IVsHierarchy hierarchy = null;
+            uint vsItemID = (uint)VSConstants.VSITEMID.Nil;
+
+            if (TryGetHierarchy(filePath, out hierarchy, out vsItemID))
+            {
+                object propertyObject;
+
+                int hr = hierarchy.GetProperty((uint)VSConstants.VSITEMID.Root, (int)__VSHPROPID.VSHPROPID_Name, out propertyObject);
+
+                if (hr == VSConstants.S_OK)
+                {
+                    projectName = propertyObject as string;
+                }
+            }
+
+            return projectName;
+        }
+
+        public static bool TryGetHierarchy(string filePath, out IVsHierarchy vsHierarchy, out uint vsItemId)
+        {
+            bool result = true;
+
+            vsHierarchy = null;
+            vsItemId = (uint)VSConstants.VSITEMID.Nil;
+
+            IVsUIShellOpenDocument vsUIShellOpenDocument = ServiceProvider.GlobalProvider.GetService(typeof(SVsUIShellOpenDocument)) as IVsUIShellOpenDocument;
+
+            IOleServiceProvider serviceProviderUnused = null;
```

Comment:
nit: you don't need this, you can just use `out _` in the method call and it will be discarded.

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/discards

Reply:
same with `docInProject`

Reply:
Nice, didn't know about these, thanks!